src/client.rs

```diff
@@ -149,21 +114,22 @@ impl<IO: AsyncRead + AsyncWrite + Unpin> EppClient<IO> {
         Ok(GreetingDocument::deserialize(&response)?.data)
     }
 
-    pub async fn transact<'c, 'e, C, E>(
+    pub async fn transact<'c, 'e, Cmd, Ext>(
         &mut self,
-        data: impl Into<RequestData<'c, 'e, C, E>>,
+        data: impl Into<RequestData<'c, 'e, Cmd, Ext>>,
         id: &str,
-    ) -> Result<Response<C::Response, E::Response>, Error>
+    ) -> Result<Response<Cmd::Response, Ext::Response>, Error>
     where
-        C: Transaction<E> + Command + 'c,
-        E: Extension + 'e,
+        Cmd: Transaction<Ext> + Command + 'c,
+        Ext: Extension + 'e,
     {
         let data = data.into();
-        let epp_xml = <C as Transaction<E>>::serialize_request(data.command, data.extension, id)?;
+        let epp_xml =
+            <Cmd as Transaction<Ext>>::serialize_request(data.command, data.extension, id)?;
 
         let response = self.connection.transact(&epp_xml).await?;
 
-        C::deserialize_response(&response)
+        Cmd::deserialize_response(&response)
```

Comment:
These changes are just for clarity right?

Reply:
There's a conflict with the `C` type argument that's now the `Connector`, so I had to rename the `C` for `Command`.